CrowsZA/functions/fn_surrenderChance.sqf

```diff
@@ -0,0 +1,137 @@
+/*/////////////////////////////////////////////////
+Author: Landric
+			   
+File: fn_surrenderChance.sqf
+Parameters:
+	unit					- (unit)
+	confrontationRadius		- (float) radius within which pointing a weapon at the unit will trigger a response
+	surrenderChance			- (float) min: 0 max: 1
+	holdFire				- (bool) will unit hold fire until confronted
+	hesitation				- (float) maximum time unit will stall before acting once confronted
+
+Return: bool        		- True if method ran successfully, false if aborted (i.e. bad params were passed)
+
+// This code inspired by CQB Interactions By Flex7103
+// https://steamcommunity.com/sharedfiles/filedetails/?id=2942202773
+
+*///////////////////////////////////////////////
+params [
+	["_unit", objNull, [objNull]],
+	["_confrontationRadius", 15, [15]],
+	["_surrenderChance", 0.33, [0.33]],
+	["_holdFire", true, [true]],
+	["_hesitation", 0, [0]]
+];
+
+
+if(isNull _unit) exitWith { false };
+
+
+// TODO: Add check if unit has already had this module applied/is already surrendered/etc
```

Comment:
I think this is a good idea to avoid unexpected behaviour. Easy way of doing it, is to set a variable on the unit with ``_unit setVariable["crowsza_surrender_chance_applied", true. true];`` 
And then just simply read it here: ``_unit getVariable["crowsza_surrender_chance_applied", false];``

Just remember to have the last ``true`` on, to broadcast the variable so other zeus can also read that variable.  

You could also add to this a little icon or text for zeus on the unit, so zeus can see its applied. Like we are drawing the Zeus RC icon. (A bit more complicated and would require a CBA options to turn it on/off probably, but could be a neat information). Easiest for now is definetly to just make it exit here if applied and give a "hint/systemChat" with "already applied" or something.  
If you want to go the draw route, you can see how RC is done here: https://github.com/Crowdedlight/Crows-Zeus-Additions/blob/main/CrowsZA/functions/fn_addZeusTextDisplayEH.sqf#L52 
And you can set a common array of units with it applied for drawing like I do in RC icon: ``missionNamespace setVariable["crowsZA_rcUnits", _rcUnits, true];``